### DISCUSSION TITLE
struct.error: unpack requires a buffer of 6906162 bytes

### DISCUSSION THREAD

**Discussion by JP-NYC** [extremecoders-re/pyinstxtractor#50] [Q&A]
I'm (a newbie) trying to look at the contents of a file that runs on a Mac in Terminal. I think is written in python because when I look at the contents in a hex editor I see some references to py and python in it.  I tried to get pyinstxtractor to run on the Mac (only using the native built in python) but couldn't so I installed python on my Windows machine.

When I run pyinstxtractor on it this is my output, not sure how to proceed. I added the pyo extension to the file.

**C:\Users\David>D:\pyinstxtractor.py d:\SA.pyo
[+] Processing d:\SA.pyo
[+] Pyinstaller version: 2.0
[+] Python version: 309
[+] Length of package: 9419143 bytes
Traceback (most recent call last):
  File "D:\pyinstxtractor.py", line 423, in <module>
    main()
  File "D:\pyinstxtractor.py", line 411, in main
    arch.parseTOC()
  File "D:\pyinstxtractor.py", line 235, in parseTOC
    struct.unpack( \
struct.error: unpack requires a buffer of 6906162 bytes**

**Comment by extremecoders-re** [comment]
This most likely looks like a modded pyinstaller executable. 
If you attach the file in question can give you a better answer after taking a look.

**Comment by JP-NYC** [comment]
Would much appreciate it if you would look at it.
Is it possible to to somehow either DM it to you or email it (9mb unzipped), so I dont have to publicly post it?

**Reply by extremecoders-re** [reply]
My email is mentioned on my GitHub profile.

**Reply by JP-NYC** [reply]
Emailed it to the address in your profile.
Thanks

**Reply by extremecoders-re** [reply]
Replied to your mail. Its extracting fine. Not modded in any way.

**Reply by JP-NYC** [reply]
1. First of all. Thank you! Thank you! Thank you! You may have saved my marriage. I was so upset with my wife for running unknown code.
2. I downloaded the extracted files you provided. There are so may files and folders. I assume much of the extracted files/folders are just common python libraries etc. That would be found in any extracted exe. I will probably play further with this just for educational purposes some time in the future (and going through all of this is probably above my pay-grade anyway), but just to confirm you looked everything over and there is nothing for me to be concerned about. The program does only what it says it does (correct)?

Than you again for all of you efforts :-)

**Reply by extremecoders-re** [reply]
I only took a quick look at the files and from that this doesn't look malicious, atleast on the surface. Yes much of the files are from standard Python files/libraries - judging from the names. However there's no way to fully guarantee that as malicious behaviour can always be hidden deep underneath. If the origin of the file is not something to worry about then I guess this is also benign.

**Comment by JP-NYC** [comment]
Hi again,

Wondering if you would be interested and what you would charge me to go through the program and just confirm that it only does what it says it does and that there is nothing more hidden in it.


I doubt that this guy is a government level master hacker so I would just need you do do this at medium level and confirm it is all on the up and up.


Thanks again for for time & effort.


Jack


________________________________

**Reply by extremecoders-re** [reply]
Hi, manually going through program to ascertain its not malicious is quite time taking. You can always use online scanners like virustotal for a quick insight.
